src/d3d9/d3d9_cursor.cpp

```diff
@@ -25,24 +25,55 @@ namespace dxvk {
   }
 
 
-  HRESULT D3D9Cursor::SetHardwareCursor(UINT XHotSpot, UINT YHotSpot, const CursorBitmap& bitmap) {
-    DWORD mask[32];
-    std::memset(mask, ~0, sizeof(mask));
+  HRESULT D3D9Cursor::SetHardwareCursor(
+          UINT               XHotSpot,
+          UINT               YHotSpot,
+          const uint8_t*     bitmap,
+          bool               cursorEmulation,
+          UINT               width,
+          UINT               height,
+          HWND               window) {
+    bool noPreviousHardwareCursor = false;
+
+    // For color icons, the hbmMask and hbmColor bitmaps
+    // are the same size, each of which is the size of the icon.
+    uint32_t maskSize = cursorEmulation ? width * height : HardwareCursorWidth * HardwareCursorHeight;
+    DWORD* mask = new DWORD[maskSize];
```

Comment:
Just use a std::vector.

Reply:
@Joshua-Ashton Done, hopefully in a right manner. Let me know if I've initialized it correctly :sweat_smile:. I was half expecting MSVC to complain about something, because why not.